CUJExecutor/dataAggregation/Utility.java

```diff
@@ -0,0 +1,236 @@
+	String start; //the time at which this clip begins in the video (in the format HH:MM:SS.sss)
+	String end; //the time at which this clip ends in the video (in the format HH:MM:SS.sss)
+
+	public clipInfo (String fldr, String timestamps) {
+		folder = fldr;
+		String[] startEnd = timestamps.split("\\s*,\\s*");
+		start = startEnd[0];
+		end = startEnd[1];
+	}
+	public String getFolder() {
+		return folder;
+	}
+	public String getStart() {
+		return start;
+	}
+
+	public String getEnd() {
+		return end;
+	}
+   }
+   
+
+
+   public static Integer sum(List<Integer> list) {
+	   int sum = 0;
+	   for (int n : list) {
+		   sum += n;
+	   }
+	   return sum;
+   }
+
+   public static Integer average(List<Integer> list) {
+	   if (list.size() == 0) return null;
+	   return (sum(list) / list.size());
+   }
+
+ /**
+   * calculates the average of each column
+   * ith entry of returned array is average of ith column of input
+   */
+    public static List<Integer> averageColumns(List<List<Integer>> mat) {
+	if (mat.size() == 0) return null;
+	List<Integer> averages = new ArrayList<>();
+        for (int j = 0; j < mat.get(0).size(); j++) {
+            int sum = 0;
+            for (int i = 0; i < mat.size(); i++) {
+                sum += mat.get(i).get(j);
+            }
+            averages.add(sum / mat.size());
+        }
+        return averages;
+    }
+
+    /**
+     * Returns the median of the provided list
+     * If list is even, returns the greater of the two middle elements
+     * if list is empty, return null
+     */
+    public static Integer median(List<Integer> list) {
+	    if (list.size() == 0) return null;
+	    Collections.sort(list);
+	    return list.get(list.size() / 2);
+	    
+	    /*if (list.size() % 2 == 1) {
+	    } else {
+		    return (list.get(list.size() / 2 - 1) + list.get(list.size() / 2)) / 2;
+	    }*/
+    }
+
+   /**
+     * calculates the median of each column
+     * ith entry of returned array is median of ith column of input
+     */
+    public static List<Integer> medianColumns(List<List<Integer>> mat) {
+        if (mat.size() == 0) return null;
+	List<Integer> medians = new ArrayList<>();
+        for (int j = 0; j < mat.get(0).size(); j++) {
+            int[] column = new int[mat.size()];
+            for (int i = 0; i < mat.size(); i++) {
+                column[i] = mat.get(i).get(j);
+            }
+            Arrays.sort(column);
+            medians.add(column[column.length / 2]);
+            /*if (column.length % 2 == 1) {
+            } else {
+                medians.add((column[column.length / 2] + column[(column.length / 2) - 1]) / 2);
+            }*/
+        }
+        return medians;
+    }	
+    
+    /**
+     * Prints str to the provided printStream in a column with the width specified by alignment
+     */
+    public static void printWithAlignment(PrintStream ps, int alignment, String str) {
+	ps.printf("%-" + (alignment + 1) + "s", str);
+    }
```

Comment:
Should have a method that returns an aligned string.  Then the caller can choose to print whatever way they like.  In other words, separate the alignment from the actual printing.